pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -2231,44 +2280,36 @@ func (cc *XController) manageQueueJob(qj *arbv1.AppWrapper, podPhaseChanges bool
 						qj.Name, time.Now().Sub(qj.Status.ControllerFirstTimestamp.Time).Seconds(), cc.qjqueue.IfExistActiveQ(qj), cc.qjqueue.IfExistUnschedulableQ(qj), qj, qj.ResourceVersion, qj.Status)
 				}
 			}
-
-			//_, err = cc.arbclients.ArbV1().AppWrappers(qj.Namespace).Update(qj)
-			//if err != nil {
-			//	return err
-			//}
 			return nil
 		}
-
-		// if !qj.Status.CanRun && qj.Status.State == arbv1.QueueJobStateEnqueued {
 		if !qj.Status.CanRun && qj.Status.State == arbv1.AppWrapperStateEnqueued {
 			cc.qjqueue.AddIfNotPresent(qj)
 			return nil
 		}
-
 		if qj.Status.CanRun && !qj.Status.IsDispatched {
 			if klog.V(10).Enabled() {
 				current_time := time.Now()
 				klog.V(10).Infof("[worker-manageQJ] XQJ %s has Overhead Before Dispatching: %s", qj.Name, current_time.Sub(qj.CreationTimestamp.Time))
-				klog.V(10).Infof("[TTime] %s, %s: WorkerBeforeDispatch", qj.Name, time.Now().Sub(qj.CreationTimestamp.Time))
+				klog.V(10).Infof("[worker-manageQJ] %s, %s: WorkerBeforeDispatch", qj.Name, time.Now().Sub(qj.CreationTimestamp.Time))
 			}
 
 			queuejobKey, _ := GetQueueJobKey(qj)
-			// agentId:=cc.dispatchMap[queuejobKey]
-			// if agentId!=nil {
-			if agentId, ok := cc.dispatchMap[queuejobKey]; ok {
-				klog.V(10).Infof("[Dispatcher Controller] Dispatched AppWrapper %s to Agent ID: %s.", qj.Name, agentId)
-				cc.agentMap[agentId].CreateJob(qj)
-				qj.Status.IsDispatched = true
-			} else {
-				klog.Errorf("[Dispatcher Controller] AppWrapper %s not found in dispatcher mapping.", qj.Name)
+			if cc.isDispatcher {
+				if agentId, ok := cc.dispatchMap[queuejobKey]; ok {
+					klog.V(10).Infof("[worker-manageQJ] Dispatched AppWrapper %s to Agent ID: %s.", qj.Name, agentId)
+					cc.agentMap[agentId].CreateJob(qj)
+					qj.Status.IsDispatched = true
+				} else {
+					klog.Errorf("[worker-manageQJ] AppWrapper %s not found in dispatcher mapping.", qj.Name)
+				}
 			}
 			if klog.V(10).Enabled() {
 				current_time := time.Now()
-				klog.V(10).Infof("[Dispatcher Controller] XQJ %s has Overhead After Dispatching: %s", qj.Name, current_time.Sub(qj.CreationTimestamp.Time))
-				klog.V(10).Infof("[TTime] %s, %s: WorkerAfterDispatch", qj.Name, time.Now().Sub(qj.CreationTimestamp.Time))
+				klog.V(10).Infof("[worker-manageQJ] XQJ %s has Overhead After Dispatching: %s", qj.Name, current_time.Sub(qj.CreationTimestamp.Time))
+				klog.V(10).Infof("[worker-manageQJ] %s, %s: WorkerAfterDispatch", qj.Name, time.Now().Sub(qj.CreationTimestamp.Time))
 			}
-
-			if _, err := cc.arbclients.ArbV1().AppWrappers(qj.Namespace).Update(qj); err != nil {
+			//FIXME use the updateStatusInEtcd
+			if _, err := cc.updateEtcd(qj, "[worker-manageQJ] -- set dispatched true"); err != nil {
```

Comment:
curious to know the reason why we have FIXME

Reply:
Because this is still outstanding, we'll remove!